packages/@lwc/engine-core/src/framework/modules/context.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import {
+    isUndefined,
+    getPrototypeOf,
+    keys,
+    getContextKeys,
+    ArrayFilter,
+    ContextEventName,
+    isTrustedContext,
+} from '@lwc/shared';
+import { type VM } from '../vm';
+import { logErrorOnce } from '../../shared/logger';
+import type { Signal } from '@lwc/signals';
+import type { RendererAPI } from '../renderer';
+import type { ShouldContinueBubbling } from '../wiring/types';
+
+type ContextProvidedCallback = (contextSignal?: Signal<unknown>) => void;
+type ContextVarieties = Map<unknown, Signal<unknown>>;
+
+class ContextConnector<T extends object> {
```

Comment:
```suggestion
class ContextConnector<Cmp extends object> {
```
`T` on the class and `T` on methods is confusing

Reply:
Agreed, I prefer C and V though